### ISSUE TITLE
Spinner component variations

### ISSUE THREAD

**Issue by cnrudd** [xh/hoist-react#2919]
We are using this [spinner component](https://github.com/xh/hoist-react/blob/develop/cmp/spinner/Spinner.js) in a client app for two job states:  PENDING and PROCESSING.

It would be nice to have a way to show a spinning gear for the PROCESSING state.

This could be accomplished in one of three ways:

1. make a new component that uses the VM optimized gear
2. modify component to take a prop option string for image name for images that are built in:  'default', 'gear', 'dotdotdot', etc
3. modify component to take the actual image used as a prop

I'm in favor of #2, since it lets us make sure app developers do not load bonkers heavy animated gifs into the component, and we could always pretty easily add more image options once the pattern is in place.

**Comment by diegoherrer4** [comment]
I can work on this

**Comment by amcclain** [comment]
Effectively implemented by the v84.0.0 Spinner rework: the component is now FontAwesome icon-based, with the icon selectable per-instance via the iconName / prefix props or globally via Spinner.defaults (e.g. iconName: 'gear' for a spinning gear). The legacy PNG remains available via usePng. That delivers the option 2 approach preferred here - a constrained, name-based variant choice rather than arbitrary app-supplied images.
